### ISSUE TITLE
`TypeAlias` with stringified builtin types

### ISSUE THREAD

**Issue by Molkree** [microsoft/pyright#4413]
**Describe the bug**
Maybe a followup to #4320, I don't see any error diagnostic when my `TypeAlias` is a builtin type wrapped in a string.

**Expected behavior**
New `typeAliasIllegalExpressionForm` error? Or "correct" type inference (should be `str`)?

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import TypeAlias

MyType: TypeAlias = "int"
```
![image](https://user-images.githubusercontent.com/12860284/210879506-e71cf4cf-b3bd-474d-8c42-4b919d2c57da.png)

**VS Code extension or command-line**
Pylance v2023.1.11, with Pyright [1.1.287](https://github.com/microsoft/pyright/releases/tag/1.1.287) included.

**Comment by Molkree** [comment]
@erictraut, my apologizes, in an effort to show the minimum reproducible example I have confused myself.
Consider this code:
```py

**Comment by erictraut** [comment]
Yeah, the `|` operator at runtime unfortunately doesn't work with strings. Pyright makes an effort to detect and report an error for this condition when using string literals with `|`, but it doesn't do this for type aliases. Doing so would require pyright to track additional information about type aliases that it doesn't currently. That's possible to do, but I don't think it's worth the effort. After all, this is a false negative case, it's an extreme edge case that very few users are ever likely to encounter, and it's something that will immediately be detected at runtime the first time you run your code.

I might consider adding support for this in the future if there's sufficient interest (in the form of upvotes on this issue), but for now, I'm going to keep this issue closed because I don't plan to add this support.
